File: calculator.go
package main

import (
	"fmt"
)

func calculate(op string, num1, num2 float64) (float64, error) {
	switch op {
	case "+":
		return num1 + num2, nil
	case "-":
		return num1 - num2, nil
	case "*":
		return num1 * num2, nil
	case "/":
		if num2 != 0 {
			return num1 / num2, nil
		}
		return 0, fmt.Errorf("error: division by zero")
	default:
		return 0, fmt.Errorf("error: unsupported operation")
	}
}

func main() {
	var num1, num2 float64
	var operation string

	fmt.Println("Simple Calculator")

	fmt.Print("Enter first number: ")
	_, err := fmt.Scan(&num1)
	if err != nil {
		fmt.Println("Invalid input for the first number.")
		return
	}

	fmt.Print("Enter operation (+, -, *, /): ")
	_, err = fmt.Scan(&operation)
	if err != nil {
		fmt.Println("Invalid input for operation.")
		return
	}

	fmt.Print("Enter second number: ")
	_, err = fmt.Scan(&num2)
	if err != nil {
		fmt.Println("Invalid input for the second number.")
		return
	}

	result, err := calculate(operation, num1, num2)
	if err != nil {
		fmt.Println(err)
	} else {
		fmt.Printf("Result: %.2f\n", result)
	}
}

File: calculator.java
import java.util.Scanner;

public class Calculator {

    // Method to perform calculations
    public static double calculate(double num1, double num2, String operation) {
        switch (operation) {
            case "+":
                return num1 + num2;
            case "-":
                return num1 - num2;
            case "*":
                return num1 * num2;
            case "/":
                if (num2 != 0) {
                    return num1 / num2;
                } else {
                    throw new ArithmeticException("Error: Division by zero");
                }
            default:
                throw new IllegalArgumentException("Error: Unsupported operation");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Simple Calculator");

        while (true) {
            try {
                System.out.print("Enter first number: ");
                double num1 = scanner.nextDouble();

                System.out.print("Enter operation (+, -, *, /): ");
                String operation = scanner.next();

                System.out.print("Enter second number: ");
                double num2 = scanner.nextDouble();

                double result = calculate(num1, num2, operation);
                System.out.println("Result: " + result);

                System.out.print("Do you want to perform another calculation? (yes/no): ");
                String continueCalc = scanner.next();
                if (!continueCalc.equalsIgnoreCase("yes")) {
                    break;
                }
            } catch (ArithmeticException e) {
                System.out.println(e.getMessage());
            } catch (IllegalArgumentException e) {
                System.out.println(e.getMessage());
            } catch (Exception e) {
                System.out.println("Invalid input. Please try again.");
                scanner.next(); // Clear the invalid input
            }
        }

        scanner.close();
        System.out.println("Goodbye!");
    }
}
